src/commonTypes.ts

```diff
@@ -51,35 +50,34 @@ export type ExtendedOptions = OptionsWithDefaults & {
 };
 
 /*
- * Page
+ * Pages
  */
-export type PageFile<FileType> = {
-  client: FileType;
-  server: FileType;
-};
-
 export type PageParams = ParsedUrlQuery;
 
 export type RouteInfo = {
   params: PageParams;
   query: PageParams;
   route: string;
+  // Page file path without extension
   pagePath: string;
   paramsNumber: number;
   resolvedUrl: string;
 };
 
 export type PageObject = RouteInfo & {
   type: 'found';
-  page: PageFile<NextPageFile>;
-  appFile: PageFile<NextAppFile>;
+  absolutePagePath: string;
+  files: MultiEnv<NextPageFiles>;
 };
 
 export type NotFoundPageObject = RouteInfo & {
   type: 'notFound';
-  appFile: PageFile<NextAppFile>;
+  absolutePagePath: string;
+  files: MultiEnv<NextErrorPageFiles>;
 };
 
+export type GenericPageObject = PageObject | NotFoundPageObject;
```

Comment:
Just a suggestion. Actually don't have good ideas:

```suggestion
export type GenericPageObject = ExistingPageObject | NotFoundPageObject;
```